lib/foreman/renderer.rb

```diff
@@ -27,26 +30,39 @@ def template_logger
     end
 
     def host_enc(*path)
+      check_host
       @enc ||= @host.info.deep_dup
       return @enc if path.compact.empty?
       enc = @enc
+      step = nil
       path.each { |step| enc = enc.fetch step }
       enc
+    rescue KeyError
+      raise(HostENCParamUndefined, _('Parameter %{name} is not set in host %{host} ENC output') % { :name => step, :host => @host })
```

Comment:
One last thing: I guess we should have `path` here, rather than `step`, as if the `step` is nested deeper in some path, it might not be clear which step exactly it is.

Reply:
I added both, from path only it would not be clear what step is missing which might be hard to find in case of big hashes